PizzaBuilder/PizzaBuilder/Program.cs

```diff
@@ -0,0 +1,229 @@
+        public WhitePizzaBulider()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+            this._product = new WhitePizza();
+        }
+        public void BulidDough()
+        {
+            this._product.Add("White Dough");
+        }
+        public void BuildSauce()
+        {
+            this._product.Add("Sauce");
+        }
+        public void BuildTopping()
+        {
+            this._product.Add("Topping");
+        }
+        public WhitePizza GetProduct()
+        {
+            WhitePizza result = this._product;
+            this.Reset();
+            return result;
+        }
+    }
+    public class WheatPizzaBulider : IBuilder
+    {
+        private WheatPizza _product = new WheatPizza();
+        public WheatPizzaBulider()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+            this._product = new WheatPizza();
+        }
+        public void BulidDough()
+        {
+            this._product.Add("Wheat Dough");
+        }
+        public void BuildSauce()
+        {
+            this._product.Add("Sauce");
+        }
+        public void BuildTopping()
+        {
+            this._product.Add("Topping");
+        }
+        public WheatPizza GetProduct()
+        {
+            WheatPizza result = this._product;
+            this.Reset();
+            return result;
+        }
+    }
+    public class WhitePizza
+    {
+        private List<object> parts = new List<object>();
+        public void Add(string part)
+        {
+            this.parts.Add(part);
+        }
+        public string listParts()
+        {
+            string str = "";
+            for (int i = 0; i < this.parts.Count; i++)
+            {
+                str += this.parts[i] + ",";
+            }
+            return "product parts:" + str;
+        }
+    }
+    public class WheatPizza
+    {
+        private List<object> parts = new List<object>();
+        public void Add(string part)
+        {
+            this.parts.Add(part);
+        }
+        public string listParts()
+        {
+            string str = "";
+            for (int i = 0; i < this.parts.Count; i++)
+            {
+                str += this.parts[i] + ",";
+            }
+            return "product parts:" + str;
+        }
+    }
+/*    public class Director
+    {
+        private IBuilder _builder;
+        public IBuilder Builder
+        {
+            set { this._builder = value; }
+        }
+        public void BuildMin()
+        {
+            this._builder.BulidDough();
+        }
+        public void BuildFull()
+        {
+            this._builder.BuildTopping();
+            this._builder.BulidDough();
+            this._builder.BuildSauce();
+        }
+    }*/
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            //client
+           // var director = new Director();
+            var White = new WhitePizzaBulider();
```

Comment:
It would be better if you used polymorphism to define builders, without creating new builder variable for each one.